llms/openai/internal/openaiclient/openaiclient.go

```diff
@@ -11,10 +11,11 @@ var ErrEmptyResponse = errors.New("empty response")
 
 // Client is a client for the OpenAI API.
 type Client struct {
```

Comment:
Possibly out of scope for this PR, but worth considering: We could define a `transport` which is used by the `Client` that sets cross-cutting headers so we can avoid duplicating that logic across multiple files i.e. `embedding.go`, `chat.go`, etc

```go
// Implements http.RoundTripper
type transport struct {
    token string
    organization string
}

func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
    req.Header.Set("Content-Type", "application/json")
    req.Header.Set("Authorization", "Bearer "+t.token)
    if t.organization != "" {
        req.Header.Set("OpenAI-Organization", t.organization)
    }
    return http.DefaultTransport.RoundTrip(req)
 } 
```

Alternatively, we could define a single `newHTTPRequest` function that sets all the headers as required.